share/lib/reports.awk

```diff
@@ -15,20 +18,45 @@ BEGIN {
   close("date +%s.%N");
 }
 {
-  ## Enter the main loop
-  # $2 -> technique
-  # $3 -> result
-  # $7 -> component
-  # $8 -> key
-  # $11 -> message
-  ##
+  is_report = 0;
+
+  if (NF > 1)
+  {
+    # $1 is the report, the rest is the message
+    # split the first part of the report
+    nf_report = split($1, r, "##|@@");
+
+    ##
+    # r[2]  -> technique
+    # r[3]  -> result
+    # r[7]  -> component
+    # r[8]  -> key
+    # r[11] -> message
+    ##
+
+    # the rest is the message
+    message = substr($0, length($1) + 3);
```

Comment:
Not if the message itself contains a "@#". The idea is to treat what comes after the first "@#" as a verbatim string.